lib/assert.js

```diff
@@ -283,6 +298,26 @@ function _deepEqual(actual, expected, strict, memos) {
   return objEquiv(actual, expected, strict, memos);
 }
 
+function setEquiv(a, b) {
+  // This behaviour will work for any sets with contents that have
+  // strict-equality. That is, it will return false if the two sets contain
+  // equivalent objects that aren't reference-equal. We could support that, but
+  // it would require scanning each pairwise set of not strict-equal items,
+  // which is O(n^2), and would get exponentially worse if sets are nested. So
+  // for now we simply won't support deep equality checking set items or map
+  // keys.
```

Comment:
... Another strong argument in favor of making it correct > fast is that it'll be more compatible with the current implementation. If someone currently has a current test that reads `assert.deepStrictEqual(new Set([{x:5}]), new Set([{x:5}]))` that test will currently pass because the sets won't be compared. With the PR in its current state, that test will start to fail.